text/0039-reduce-module-size.md

```diff
@@ -0,0 +1,550 @@
+   class from the packages in step 1. Since `aws-cdk-lib`'s source code is
+   auto-generated from the v1 submodules, this could be an autogenerated
+   wrapper. We will need to do something clever here to make sure customers do
+   not have IDE or compilation errors before the CDK CLI has an opportunity to
+   download and install the packages.
+3. Modify the CDK CLI to verify that the correct packages are available in the
+   `node_modules` directory during synthesis. If they are not available,
+   download them from npm.
+4. We need to keep the dependencies up-to-date to get security updates and new
+   features. Each Lambda Layer will need to be treated slightly differently.
+    1. `@aws-cdk/lambda-layer-kubectl` - This one has two dependencies to keep
+       updated. All updates described below will result in a minor version bump
+       to this package.
+        1. [kubectl](https://kubernetes.io/docs/tasks/tools/#kubectl)
+            1. Kubernetes users will need to be able to choose the minor version
+               of kubectl that they want to use. And the aws-eks module will
+               need to as well.
+            2. The public API for this package will have a `KubectlV1_20Layer`
+               class for each currently supported minor version of kubernetes.
+            3. A weekly automated task will check the [Kubernetes
+               API](https://dl.k8s.io/release/stable.txt) for a new patch
+               version of each currently supported minor version, and
+               automatically update the existing `Layer` class to use the new
+               patch version. This update will be performed with an
+               auto-approved PR.
+            4. A weekly automated task will check for new supported minor
+               versions of kubectl. If there is one, a PR will be automatically
+               created which creates a new `Layer` class in the API for the new
+               minor version. A human will need to review this one.
+        2. [helm](https://helm.sh/) is also included in each Lambda Layer in
+           this package
+            1. Helm has a somewhat complicated version support policy documented
+               [here](https://helm.sh/docs/topics/version_skew/).
+            2. For each `KubectlVX_YLayer` class, we will include the latest
+               available version of Helm which is compatible with that version
+               of Kubernetes.
+            3. A weekly automated task will update the minor and patch versions
+               of helm within the supported ranges.
+            4. The automated task that checks for new supported minor versions
+               of kubectl, will also attempt to figure out what is the latest
+               version of helm compatible with that version of kubectl. This
+               will be verified by a human in PR review to make sure it is
+               correct.
+    2. `@aws-cdk/lambda-layer-aws-cli` - Major versions 1 and 2 will both be
```

Comment:
Also here why not 2 packages?

Reply:
Good point here as well! Updated the doc :)